examples/stress_tests/many_gizmos.rs

```diff
@@ -0,0 +1,110 @@
+use std::f32::consts::TAU;
+
+use bevy::{
+    diagnostic::{Diagnostics, FrameTimeDiagnosticsPlugin},
+    prelude::*,
+    window::PresentMode,
+};
+
+const SYSTEM_COUNT: u32 = 10;
+
+fn main() {
+    let mut app = App::new();
+    app.add_plugins(DefaultPlugins.set(WindowPlugin {
+        primary_window: Some(Window {
+            title: "Many Debug Lines".to_string(),
+            present_mode: PresentMode::AutoNoVsync,
+            ..default()
+        }),
+        ..default()
+    }))
+    .add_plugin(FrameTimeDiagnosticsPlugin::default())
+    .insert_resource(Config {
+        line_count: 50_000,
+        fancy: false,
+    })
+    .insert_resource(GizmoConfig {
+        on_top: false,
+        ..default()
+    })
+    .add_startup_system(setup)
+    .add_system(input)
+    .add_system(ui_system);
+
+    for _ in 0..SYSTEM_COUNT {
+        app.add_system(system);
+    }
+
+    app.run();
+}
+
+#[derive(Resource, Debug)]
+struct Config {
+    line_count: u32,
+    fancy: bool,
+}
+
+fn input(mut config: ResMut<Config>, input: Res<Input<KeyCode>>) {
+    if input.just_pressed(KeyCode::Up) {
+        config.line_count += 10_000;
+    }
+    if input.just_pressed(KeyCode::Down) {
+        config.line_count = config.line_count.saturating_sub(10_000);
+    }
+    if input.just_pressed(KeyCode::Space) {
+        config.fancy = !config.fancy;
+    }
+}
+
+fn system(config: Res<Config>, time: Res<Time>, mut draw: Gizmos) {
+    if !config.fancy {
+        for _ in 0..(config.line_count / SYSTEM_COUNT) {
+            draw.line(Vec3::NEG_Y, Vec3::Y, Color::BLACK);
+        }
```

Comment:
It would be clearer something was going on if the text was rendering :sweat_smile:
I could also make the fancy mode the default (press space to see it)